sql/src/influxql/planner.rs

```diff
@@ -1,34 +1,102 @@
 // Copyright 2023 CeresDB Project Authors. Licensed under Apache-2.0.
 
-//! Influxql planner
+//! InfluxQL planner
 
-use std::sync::Arc;
+use std::{collections::HashMap, sync::Arc};
 
 use common_util::error::BoxError;
-use influxql_logical_planner::planner::InfluxQLToLogicalPlan;
+use datafusion::{error::DataFusionError, sql::planner::ContextProvider};
+use datafusion_expr::TableSource;
+use influxql_logical_planner::plan::{
+    ceresdb_schema_to_influxdb, InfluxQLToLogicalPlan, SchemaProvider,
+};
 use influxql_parser::{
     common::{MeasurementName, QualifiedMeasurementName},
     select::{MeasurementSelection, SelectStatement},
     show_measurements::ShowMeasurementsStatement,
     statement::Statement as InfluxqlStatement,
 };
+use influxql_schema::Schema;
 use snafu::{ensure, ResultExt};
+use table_engine::table::TableRef;
 
 use crate::{
-    influxql::{error::*, provider::InfluxSchemaProviderImpl},
+    influxql::error::*,
     plan::{Plan, QueryPlan, QueryType, ShowPlan, ShowTablesPlan},
     provider::{ContextProviderAdapter, MetaProvider},
 };
 
-pub const CERESDB_MEASUREMENT_COLUMN_NAME: &str = "ceresdb::measurement";
+// Same with iox
+pub const CERESDB_MEASUREMENT_COLUMN_NAME: &str = "iox::measurement";
 
-pub(crate) struct Planner<'a, P: MetaProvider> {
+// Port from https://github.com/ceresdb/influxql/blob/36fc4d873e/iox_query_influxql/src/frontend/planner.rs#L28
+struct InfluxQLSchemaProvider<'a, P: MetaProvider> {
     context_provider: ContextProviderAdapter<'a, P>,
+    // TODO: avoid load all tables.
+    // if we can ensure `table_names` is only called once, then load tables lazily is better.
+    tables: HashMap<String, (Arc<dyn TableSource>, Schema)>,
+}
+
+impl<'a, P: MetaProvider> SchemaProvider for InfluxQLSchemaProvider<'a, P> {
+    fn get_table_provider(&self, name: &str) -> datafusion::error::Result<Arc<dyn TableSource>> {
+        self.tables
+            .get(name)
+            .map(|(t, _)| Arc::clone(t))
+            .ok_or_else(|| DataFusionError::Plan(format!("measurement does not exist: {name}")))
+    }
+
+    fn get_function_meta(&self, name: &str) -> Option<Arc<datafusion_expr::ScalarUDF>> {
+        self.context_provider.get_function_meta(name)
+    }
+
+    fn get_aggregate_meta(&self, name: &str) -> Option<Arc<datafusion_expr::AggregateUDF>> {
+        self.context_provider.get_aggregate_meta(name)
+    }
+
+    fn table_names(&self) -> Vec<&'_ str> {
+        self.tables.keys().map(|k| k.as_str()).collect::<Vec<_>>()
+    }
+
+    fn table_schema(&self, name: &str) -> Option<Schema> {
+        self.tables.get(name).map(|(_, s)| s.clone())
+    }
+}
+
+pub(crate) struct Planner<'a, P: MetaProvider> {
+    schema_provider: InfluxQLSchemaProvider<'a, P>,
+}
+
+fn convert_influxql_schema(ceresdb_schema: common_types::schema::Schema) -> Result<Schema> {
```

Comment:
How about `convert_influxql_schema`?

Reply:
Isn't this same with mine?

Reply:
Oh sorry... I want to say `convert_to_influxql_schema`.

Reply:
Will fix this in next PR